### PR TITLE
Vehicle and Vehicle Model

### PR DESCRIPTION
**Pull request by jmcanterafonseca**

### CODE REVIEW

**Comment by jmcanterafonseca** [comment]
@gtorodelvalle r?

File: Vehicle/Vehicle/doc/spec.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    + Optional
+
++ `location` : Vehicle's last known location represented by a GeoJSON Point. 
+    + Attribute type: `geo:json`.
+    + Normative References: [https://tools.ietf.org/html/rfc7946](https://tools.ietf.org/html/rfc7946)
+    + Attribute metadata:
+        + `timestamp`: Timestamp which captures when the vehicle was at that location.
+        This value can also appear as a FIWARE [TimeInstant](https://github.com/telefonicaid/iotagent-node-lib/blob/develop/README.md#TimeInstant)
+            + Type: [DateTime](http://schema.org/DateTime) or `ISO8601` (legacy).   
+    + Mandatory
```
Comment (jmcanterafonseca) [inline Vehicle/Vehicle/doc/spec.md]: well, we can leave it as optional, certainly, although I don't know if a vehicle not located makes sense at all :)

File: Vehicle/Vehicle/doc/spec.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+            + Type: [DateTime](http://schema.org/DateTime) or `ISO8601` (legacy).
+    + Default unit: Kilograms
+    + Optional
+
++ `vehicleIdentificationNumber` : The Vehicle Identification Number (VIN) is a unique serial number used by the automotive industry
+to identify individual motor vehicles.
+    + Normative References: [https://schema.org/vehicleIdentificationNumber](https://schema.org/vehicleIdentificationNumber)
+    + Mandatory if `vehiclePlateIdentifier` is not defined.
+    
++ `vehiclePlateIdentifier` : n identifier or code displayed on a vehicle registration plate attached to the vehicle used for official identification purposes.
```
Comment (jmcanterafonseca) [inline Vehicle/Vehicle/doc/spec.md]: yes `An` should be

File: Vehicle/Vehicle/doc/spec.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
++ `vehicleConfiguration` : A short text indicating the configuration of the vehicle, e.g. '5dr hatchback ST 2.5 MT 225 hp' or 'limited edition'.
+    + Normative References: [https://schema.org/vehicleConfiguration](https://schema.org/vehicleConfiguration)
+    + Optional
+
++ `color` : Vehicle's color
+    + Normative References: [https://schema.org/color](https://schema.org/color)
+    + Optional
+
++ `owner` : Vehicle's owner.
```
Comment (jmcanterafonseca) [inline Vehicle/Vehicle/doc/spec.md]: good point

File: Vehicle/Vehicle/doc/spec.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    + Attribute Type: [Text](https://schema.org/Text)
+    + Optional
+
++ `serviceProvided` : Service(s) provided by (or associated to) the vehicle.
```
Comment (jmcanterafonseca) [inline Vehicle/Vehicle/doc/spec.md]: yup

File: Vehicle/VehicleModel/doc/spec.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    + Optional
+
++ `weight` : Vehicle's weight.
+    + Normative References: [https://schema.org/weight](https://schema.org/weight)
+    + Optional
+
++ `vehicleEngine` : Information about the engine or engines of the vehicle.
+    + Normative References: [https://schema.org/vehicleEngine](https://schema.org/vehicleEngine)
+    + Optional
+    + Note: This property could be at vehicle level as well.
```
Comment (jmcanterafonseca) [inline Vehicle/VehicleModel/doc/spec.md]: well there can be a tuned vehicle ... which overwrites the default engine's model ;)

File: Vehicle/Vehicle/doc/spec.md (hunk 1 above)
Comment (jmcanterafonseca) [inline Vehicle/Vehicle/doc/spec.md]: fixed dd81894fe72dfd9c298fcca4590711f9eb6d6535

File: Vehicle/Vehicle/doc/spec.md (hunk 2 above)
Comment (jmcanterafonseca) [inline Vehicle/Vehicle/doc/spec.md]: fixed dd81894fe72dfd9c298fcca4590711f9eb6d6535

File: Vehicle/Vehicle/doc/spec.md (hunk 3 above)
Comment (jmcanterafonseca) [inline Vehicle/Vehicle/doc/spec.md]: fixed dd81894fe72dfd9c298fcca4590711f9eb6d6535

File: Vehicle/Vehicle/doc/spec.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    Or any other value needed by an specific application. 
+
++ `vehicleSpecialUsage` : Indicates whether the vehicle is been used for special purposes, like commercial rental,
+driving school, or as a taxi. The legislation in many countries requires this information to be revealed when offering a car for sale.
+    + Normative References: [https://auto.schema.org/vehicleSpecialUsage](https://auto.schema.org/vehicleSpecialUsage)
+    + Optional
+    
++ `vehicleType` : Type of vehicle.
+    + Attribute type: [Text](https://schema.org/Text)
+   + Allowed Values: The following values defined by *VehicleTypeEnum*,
```
Comment (jmcanterafonseca) [inline Vehicle/Vehicle/doc/spec.md]: fixed fixed dd81894fe72dfd9c298fcca4590711f9eb6d6535

File: Vehicle/Vehicle/doc/spec.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+    [DATEX 2 version 2.3](http://www.datex2.eu/sites/www.datex2.eu/files/DATEXIISchema_2_2_2_1.zip):
+        + (`agriculturalVehicle`, `bicycle`, `bus`, `car`, `caravan`,
```
Comment (jmcanterafonseca) [inline Vehicle/Vehicle/doc/spec.md]: fixed fixed dd81894fe72dfd9c298fcca4590711f9eb6d6535

File: Vehicle/Vehicle/doc/spec.md [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+responsible, district, neighbourhood, etc.
+    + Attribute type: [Text](https://schema.org/Text)
+    + Optional
+    
++ `status` : Vehicle status 
+    + One of (`parked`, `onRoute`, `outOfOrder`, `stopped`)
+    + Attribute type: [Text](https://schema.org/Text)
+    + Optional
+
++ `dateModified` : Last update timestamp of this entity
```
Comment (jmcanterafonseca) [inline Vehicle/Vehicle/doc/spec.md]: fixed fixed dd81894fe72dfd9c298fcca4590711f9eb6d6535

File: Vehicle/Vehicle/doc/spec.md [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+    + Attribute type: [DateTime](https://schema.org/DateTime)
+    + Optional
+
++ `dateCreated` : Creation timestamp of this entity
```
Comment (jmcanterafonseca) [inline Vehicle/Vehicle/doc/spec.md]: fixed fixed dd81894fe72dfd9c298fcca4590711f9eb6d6535

File: Vehicle/VehicleModel/doc/spec.md [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+    + Normative References: [https://schema.org/name](https://schema.org/name)
+    + Mandatory
+
++ `description` : Vehicle model description. 
+    + Normative References: [https://schema.org/description](https://schema.org/description)
+    + Optional
+    
++ `vehicleType` : Type of vehicle.
+    + Attribute type: [Text](https://schema.org/Text)
+   + Allowed Values: The following values defined by *VehicleTypeEnum*,
```
Comment (jmcanterafonseca) [inline Vehicle/VehicleModel/doc/spec.md]: fixed dd81894fe72dfd9c298fcca4590711f9eb6d6535

File: Vehicle/VehicleModel/doc/spec.md (hunk 5 above)
Comment (jmcanterafonseca) [inline Vehicle/VehicleModel/doc/spec.md]: NTC

File: Vehicle/VehicleModel/doc/spec.md [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+    
++ `url` : URL which provides a description of this vehicle model.
+    + Normative References: [https://schema.org/url](https://schema.org/url)
+    + Optional
+    
++ `image`: Image which depicts this vehicle model.
+    + Normative References: [https://schema.org/image](https://schema.org/image)
+    + Optional
+
++ `dateModified` : Last update timestamp of this entity
```
Comment (jmcanterafonseca) [inline Vehicle/VehicleModel/doc/spec.md]: fixed dd81894fe72dfd9c298fcca4590711f9eb6d6535

File: Vehicle/doc/introduction.md [hunk 12]
```diff
@@ -0,0 +1,16 @@
+# Vehicle Harmonized Data Models
+
+These data models describe the main entities involved with smart applications that deal with vehicle entities.
+When feasible, references to existing schema.org entity types and attributes are included.
+
+These models have been devised to be as generic as possible, thus allowing to deal with different scenarios: 
+
++ Private Vehicles. 
++ Public Vehicles (Buses, Trains, etc.).  
++ Municipal Vehicles
```
Comment (jmcanterafonseca) [inline Vehicle/doc/introduction.md]: fixed dd81894fe72dfd9c298fcca4590711f9eb6d6535

File: Vehicle/doc/introduction.md [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+
+The main entities identified are:
+
++ [Vehicle](../Vehicle/doc/spec.md) .It represents a vehicle with all its individual characteristics.
```
Comment (jmcanterafonseca) [inline Vehicle/doc/introduction.md]: fixed dd81894fe72dfd9c298fcca4590711f9eb6d6535

File: Vehicle/VehicleModel/doc/spec.md (hunk 5 above)
Comment (gtorodelvalle) [inline Vehicle/VehicleModel/doc/spec.md]: Niiice!!! :D

**Pull request merged**